### ISSUE TITLE
shell-init misbehaves when the calling shell is not $SHELL

### ISSUE THREAD

**Issue by tgyurci** [daveadams/go-rapture#5]
When the calling shell is different from the one specified in `$SHELL` then `shell-init` will generate wrong shell code which can not be eval'd.

The current case is: `fish` is specified as a Terminal start command and the user login shell is not modified, so `$SHELL` is `/bin/bash` but fish shell code is expected.  

A possible solution would be to add flags to `shell-init` (-f, -s, -z) to force a specific shell type.

**Comment by daveadams** [comment]
Good idea, and shouldn't be too hard to add. I'll think about the best implementation a bit first, but I'll try to integrate this into the next release. Thanks for the ticket.
